src/main/java/com/blackducksoftware/integration/jira/config/controller/BlackDuckConfigActions.java

```diff
@@ -0,0 +1,335 @@
+            serverConfigBuilder.setUrl(newConfig.getHubUrl());
+            serverConfigBuilder.setTimeout(newConfig.getTimeout());
+            serverConfigBuilder.setApiToken(newConfig.getApiToken());
+            serverConfigBuilder.setUsername(newConfig.getUsername());
+            serverConfigBuilder.setPassword(newConfig.getPassword());
+            serverConfigBuilder.setPasswordLength(newConfig.getPasswordLength());
+            serverConfigBuilder.setTrustCert(newConfig.getTrustCert());
+
+            serverConfigBuilder.setProxyHost(newConfig.getHubProxyHost());
+            serverConfigBuilder.setProxyPort(newConfig.getHubProxyPort());
+            serverConfigBuilder.setProxyUsername(newConfig.getHubProxyUser());
+            serverConfigBuilder.setProxyPassword(newConfig.getHubProxyPassword());
+            serverConfigBuilder.setProxyPasswordLength(newConfig.getHubProxyPasswordLength());
+            serverConfigBuilder.setProxyIgnoredHosts(newConfig.getHubNoProxyHosts());
+
+            final HubServerConfig serverConfig = serverConfigBuilder.build();
+            try (final BlackduckRestConnection restConnection = serverConfig.createRestConnection(logger)) {
+                restConnection.connect();
+            } catch (final IntegrationException | IOException e) {
+                if (e.getMessage().toLowerCase().contains("unauthorized")) {
+                    newConfig.setApiTokenError("Invalid credential(s) for: " + serverConfig.getHubUrl());
+                } else {
+                    newConfig.setTestConnectionError(e.toString());
+                }
+            }
+            return newConfig;
+        }
+    }
+
+    BlackDuckServerConfigSerializable getUnMaskedConfig(final BlackDuckServerConfigSerializable currentConfig, final PluginSettings settings) {
+        final BlackDuckServerConfigSerializable newConfig = new BlackDuckServerConfigSerializable(currentConfig);
+
+        if (StringUtils.isNotBlank(newConfig.getApiToken()) && newConfig.isApiTokenMasked()) {
+            newConfig.setApiToken(getValue(settings, BlackDuckConfigKeys.CONFIG_BLACKDUCK_API_TOKEN));
+        }
+        if (StringUtils.isNotBlank(newConfig.getPassword()) && newConfig.isPasswordMasked()) {
+            newConfig.setPassword(getValue(settings, BlackDuckConfigKeys.CONFIG_BLACKDUCK_PASS));
+            newConfig.setPasswordLength(NumberUtils.toInt(getValue(settings, BlackDuckConfigKeys.CONFIG_BLACKDUCK_PASS_LENGTH)));
+        }
+        if (StringUtils.isNotBlank(newConfig.getHubProxyPassword()) && newConfig.isProxyPasswordMasked()) {
+            newConfig.setHubProxyPassword(getValue(settings, BlackDuckConfigKeys.CONFIG_PROXY_PASS));
+            newConfig.setHubProxyPasswordLength(NumberUtils.toInt(getValue(settings, BlackDuckConfigKeys.CONFIG_PROXY_PASS_LENGTH)));
+        }
+
+        return newConfig;
+    }
+
+    // This method must be "package protected" to avoid synthetic access
```

Comment:
I think this method can be private (and the comment can be deleted) now since it's not being referenced from an anonymous class.